rs/sns/treasury_manager/mock/src/main.rs

```diff
@@ -0,0 +1,166 @@
+use ic_canister_log::{declare_log_buffer, log};
+use ic_cdk::{init, post_upgrade, query, update};
+use sns_treasury_manager::{
+    AuditTrail, AuditTrailRequest, BalancesRequest, DepositRequest, TreasuryManager,
+    TreasuryManagerArg, TreasuryManagerResult, WithdrawRequest,
+};
+use std::time::Duration;
+
+const PURPOSE: &str = "This is a template canister that is not intended to be used directly.";
+
+const RUN_PERIODIC_TASKS_INTERVAL: Duration = Duration::from_secs(60 * 60); // 1 hour
+
+pub(crate) struct MockTreasuryManager {}
+
+impl TreasuryManager for MockTreasuryManager {
+    async fn withdraw(&mut self, _request: WithdrawRequest) -> TreasuryManagerResult {
+        unimplemented!("{}", PURPOSE)
+    }
+
+    async fn deposit(&mut self, _request: DepositRequest) -> TreasuryManagerResult {
+        unimplemented!("{}", PURPOSE)
+    }
+
+    fn audit_trail(&self, _request: AuditTrailRequest) -> AuditTrail {
+        unimplemented!("{}", PURPOSE)
+    }
+
+    fn balances(&self, _request: BalancesRequest) -> TreasuryManagerResult {
+        unimplemented!("{}", PURPOSE)
+    }
+
+    async fn refresh_balances(&mut self) {
+        unimplemented!("{}", PURPOSE)
+    }
+
+    async fn issue_rewards(&mut self) {
+        unimplemented!("{}", PURPOSE)
+    }
+}
+
+declare_log_buffer!(name = LOG, capacity = 100);
+
+fn log(msg: &str) {
+    let msg = format!("[MockTreasuryManager] {}", msg);
+    ic_cdk::print(&msg);
+    log!(LOG, "{}", msg);
+}
+
+fn canister_state() -> MockTreasuryManager {
+    MockTreasuryManager {}
+}
+
+fn check_access() {
```

Comment:
should this be called something like "check caller is controller" as this is the only access that seems to be allowed?